File: bindings/lua/Polycode/RenderThreadDebugInfo.lua
class "RenderThreadDebugInfo"


function RenderThreadDebugInfo:__getvar(name)
	if name == "buffersProcessed" then
		return Polycode.RenderThreadDebugInfo_get_buffersProcessed(self.__ptr)
	elseif name == "drawCallsProcessed" then
		return Polycode.RenderThreadDebugInfo_get_drawCallsProcessed(self.__ptr)
	elseif name == "timeTaken" then
		return Polycode.RenderThreadDebugInfo_get_timeTaken(self.__ptr)
	end
end

function RenderThreadDebugInfo:__setvar(name,value)
	if name == "buffersProcessed" then
		Polycode.RenderThreadDebugInfo_set_buffersProcessed(self.__ptr, value)
		return true
	elseif name == "drawCallsProcessed" then
		Polycode.RenderThreadDebugInfo_set_drawCallsProcessed(self.__ptr, value)
		return true
	elseif name == "timeTaken" then
		Polycode.RenderThreadDebugInfo_set_timeTaken(self.__ptr, value)
		return true
	end
	return false
end
function RenderThreadDebugInfo:__delete()
	if self then Polycode.delete_RenderThreadDebugInfo(self.__ptr) end
end

File: bindings/javascript/Polycode/RenderThreadDebugInfo.js
function RenderThreadDebugInfo() {
	if(arguments[0] != "__skip_ptr__") {
		this.__ptr = Polycode.RenderThreadDebugInfo()
	}
	Object.defineProperties(this, {
		'buffersProcessed': { enumerable: true, configurable: true, get: RenderThreadDebugInfo.prototype.__get_buffersProcessed, set: RenderThreadDebugInfo.prototype.__set_buffersProcessed},
		'drawCallsProcessed': { enumerable: true, configurable: true, get: RenderThreadDebugInfo.prototype.__get_drawCallsProcessed, set: RenderThreadDebugInfo.prototype.__set_drawCallsProcessed},
		'timeTaken': { enumerable: true, configurable: true, get: RenderThreadDebugInfo.prototype.__get_timeTaken, set: RenderThreadDebugInfo.prototype.__set_timeTaken}
	})
}

RenderThreadDebugInfo.prototype.__get_buffersProcessed = function() {
	return Polycode.RenderThreadDebugInfo__get_buffersProcessed(this.__ptr)
}

RenderThreadDebugInfo.prototype.__set_buffersProcessed = function(val) {
	Polycode.RenderThreadDebugInfo__set_buffersProcessed(this.__ptr, val)
}

RenderThreadDebugInfo.prototype.__get_drawCallsProcessed = function() {
	return Polycode.RenderThreadDebugInfo__get_drawCallsProcessed(this.__ptr)
}

RenderThreadDebugInfo.prototype.__set_drawCallsProcessed = function(val) {
	Polycode.RenderThreadDebugInfo__set_drawCallsProcessed(this.__ptr, val)
}

RenderThreadDebugInfo.prototype.__get_timeTaken = function() {
	return Polycode.RenderThreadDebugInfo__get_timeTaken(this.__ptr)
}

RenderThreadDebugInfo.prototype.__set_timeTaken = function(val) {
	Polycode.RenderThreadDebugInfo__set_timeTaken(this.__ptr, val)
}

Duktape.fin(RenderThreadDebugInfo.prototype, function (x) {
	if (x === RenderThreadDebugInfo.prototype) {
		return;
	}
	Polycode.RenderThreadDebugInfo__delete(x.__ptr)
})
